scanner.go

```diff
@@ -278,14 +249,26 @@ func (s *scanner) update(resp *pb.ScanResponse, region hrpc.RegionInfo) {
 	}
 }
 
-// shouldClose check if this scanner should be closed and should stop fetching new results
-func (s *scanner) shouldClose(resp *pb.ScanResponse, region hrpc.RegionInfo) bool {
+func (s *scanner) Close() error {
+	if s.closed {
+		return errors.New("scanner has already been closed")
+	}
+	s.closed = true
+	if !s.isRegionScannerClosed() {
```

Comment:
is this check needed as isRegionScannerClosed is already idempotent?

Reply:
will remove.